robot-name/damoreno/robot_name.go

```diff
@@ -0,0 +1,40 @@
+package robotname
+
+import "fmt"
+
+type Robot struct {
+	Id string
+}
+
+var availableRobotNames = func() map[string]struct{} {
+	names := make(map[string]struct{})
+
+	for firstLetter := 'A'; firstLetter <= 'Z'; firstLetter++ {
+		for secondLetter := 'A'; secondLetter <= 'Z'; secondLetter++ {
+			for number := 0; number < 1000; number++ {
+				robotName := fmt.Sprintf("%c%c%03d", firstLetter, secondLetter, number)
+				names[robotName] = struct{}{}
```

Comment:
What does `struct{}{}` do, I mean what's the meaning in your code?

Reply:
“names” is a set (which is a hack in Go implemented with a map).  The code is only interested in storing the value with robotName (as a key), and does not care about any value.  So a type with no fields, so no value, is used as the map value as it takes no space.

Reply:
Thanks for the insight @fnasser